DotNetDBF/MemoValue.cs

```diff
@@ -1,167 +1,156 @@
-                            string tString;
-                            var tStringBuilder = new StringBuilder();
-                            int tIndex;
-                            var tSoftReturn = _base.CharEncoding.GetString(new byte[] {0x8d, 0x0a});
+                            reader.BaseStream.Seek(_block * _base.BlockSize, SeekOrigin.Begin);
+                            string tempString;
+                            StringBuilder sb = new StringBuilder();
+                            int index;
+                            string softReturn = _base.CharEncoding.GetString(new byte[] { 0x8d, 0x0a });
 
                             do
                             {
-                                var tData = reader.ReadBytes(_base.BlockSize);
-                               
-                                tString = _base.CharEncoding.GetString(tData);
-                                tIndex = tString.IndexOf(MemoTerminator);
-                                if (tIndex != -1)
-                                    tString = tString.Substring(0, tIndex);
-                                tStringBuilder.Append(tString);
-                            } while (tIndex == -1);
-                            _value = tStringBuilder.ToString().Replace(tSoftReturn,String.Empty);
+                                byte[] data = reader.ReadBytes(_base.BlockSize);
+
+                                tempString = _base.CharEncoding.GetString(data);
+                                index = tempString.IndexOf(MemoTerminator);
+                                if (index != -1)
+                                    tempString = tempString.Substring(0, index);
+                                sb.Append(tempString);
+                            } while (index == -1);
+
+                            sb.Replace(softReturn, String.Empty);
+                            _value = sb.ToString();
                         }
                         _loaded = true;
                     }
 
                     return _value;
                 }
-            }set
+            }
+            set
             {
                 lock (_lockName)
                 {
                     _new = true;
-
-                   
                     _value = value;
                 }
 
             }
         }
+        #endregion
+
+        internal void Write(DBFWriter dbfWriter)
+        {
+            lock (_lockName)
+            {
+                if (!_new)
+                    return;
+                if (string.IsNullOrEmpty(dbfWriter.DataMemoLoc))
+                    throw new Exception("No Memo Location Set");
+
+                FileStream fs = File.Open(dbfWriter.DataMemoLoc, FileMode.OpenOrCreate, FileAccess.ReadWrite);
+
+                // before proceeding check whether the passed in File object
+                // is an empty/non-existent file or not.
+
+                using (BinaryWriter writer = new BinaryWriter(fs, dbfWriter.CharEncoding))

```

Comment:
C# coding conventions always use implicit typing (aka `var`)  when the type of the variable is obvious.